x-pack/filebeat/module/suricata/eve/ingest/pipeline.json

```diff
@@ -19,83 +19,105 @@
-      { "field": "ecs.event.type"
+      { "field": "event.type"
       , "value": "{{suricata.eve.event_type}}"
       }
     }
 
   , { "set":
-      { "field": "ecs.user_agent.raw"
+      { "field": "user_agent.original"
       , "value": "{{suricata.eve.http.http_user_agent}}"
       }
     }
   , { "user_agent":
-      { "field": "ecs.user_agent.raw"
-      , "target_field": "ecs.user_agent"
+      { "field": "user_agent.original"
+      , "target_field": "user_agent"
       , "ignore_missing": true
       }
     }
   , { "rename":
-      { "field": "ecs.user_agent.os"
-      , "target_field": "ecs.user_agent.temp_os"
+      { "field": "user_agent.os"
+      , "target_field": "user_agent.temp_os"
       , "ignore_missing": true
       }
     }
   , { "rename":
-      { "field": "ecs.user_agent.temp_os"
-      , "target_field": "ecs.user_agent.os.full_name"
+      { "field": "user_agent.temp_os"
+      , "target_field": "user_agent.os.full_name"
       , "ignore_missing": true
       }
     }
   , { "rename":
-      { "field": "ecs.user_agent.os_name"
-      , "target_field": "ecs.user_agent.os.name"
+      { "field": "user_agent.os_name"
+      , "target_field": "user_agent.os.name"
       , "ignore_missing": true
       }
     }
   , { "rename":
-      { "field": "ecs.user_agent.os_version"
-      , "target_field": "ecs.user_agent.os.version"
+      { "field": "user_agent.os_version"
+      , "target_field": "user_agent.os.version"
       , "ignore_missing": true
       }
     }
   , { "rename":
-      { "field": "ecs.user_agent.os_major"
-      , "target_field": "ecs.user_agent.os.major"
+      { "field": "user_agent.os_major"
+      , "target_field": "user_agent.os.major"
       , "ignore_missing": true
       }
     }
   , { "rename":
-      { "field": "ecs.user_agent.os_minor"
-      , "target_field": "ecs.user_agent.os.minor"
+      { "field": "user_agent.os_minor"
+      , "target_field": "user_agent.os.minor"
       , "ignore_missing": true
       }
     }
-
-  , { "geoip":
+  , { "rename":
       { "field": "suricata.eve.src_ip"
-      , "target_field": "ecs.source.geo"
+      , "target_field": "source_ecs.ip"
       , "ignore_missing": true
       }
     }
-  , { "geoip":
+  , { "rename":
       { "field": "suricata.eve.dest_ip"
-      , "target_field": "ecs.destination.geo"
+      , "target_field": "destination.ip"
       , "ignore_missing": true
       }
     }
+
   , { "geoip":
-      { "field": "suricata.eve.src_ip"
-      , "target_field": "suricata.eve.src_geo"
+      { "field": "source_ecs.ip"
+      , "target_field": "source_ecs.geo"
       , "ignore_missing": true
       }
     }
   , { "geoip":
-      { "field": "suricata.eve.dest_ip"
-      , "target_field": "suricata.eve.dest_geo"
+      { "field": "destination.ip"
+      , "target_field": "destination.geo"
       , "ignore_missing": true
       }
     }
+
+  , { "set":
+      { "field": "http.request.method"
+      , "value": "{{suricata.eve.http.http_method}}"
```

Comment:
The `set` operations I had done initially were on fields that were always present. How does it behave for missing fields? Does it create an empty `http.request.method ` field? If that's the case, we must find a way that doesn't create empty fields. Because the mappings you just added are for `http` events, but you still have `dns`, `tls`, `ftp`, `smtp` and a whole bunch more to map. We can't end up with all of them defined (and empty) on each event :-)

Reply:
All of these mappings remind me that we should likely add a new processor that can do mass rename/copies with light type coercion (e.g. string to int or float, mostly). I could have sworn I had opened an issue for this. Because this is very painful at the moment.

I wonder whether Beats or Ingest Node would be a better place to have this. Perhaps Ingest Node? @adriansr @andrewkroh @tsg What do you think?

Reply:
I was planning to keep the original suricata event verbatim. Removing `src_ip` and `dest_ip` was a mistake, good catch.

I understand that `src_geo` and `dest_geo` were created by the pipeline and not included in the original event, so now that it's storing this information into source(_ecs) and destination, I'm removing them from suricata.eve. Makes sense?